Library/Formula/open-mpi.rb

```diff
@@ -27,7 +29,7 @@ class OpenMpi < Formula
   conflicts_with "lcdf-typetools", :because => "both install same set of binaries."
 
   depends_on :java => :build
-  depends_on :fortran => :recommended
+  depends_on :fortran if build.with?("fortran")
```

Comment:
This is really just a trick to move the evaluation to the formula, before the requirement has been transformed to a dependency with a different name, making the option handling in `brew install` fail to connect the `--with{,out}-fortran` (or its absence) with the `gcc` dependency.

This PR is really only here to check with the CI if the workaround has the desired effect.